compiler/optimizer/LoopReplicator.cpp

```diff
@@ -314,15 +314,17 @@ bool TR_LoopReplicator::isWellFormedLoop(TR_RegionStructure *region, TR_Structur
 
    if (_maxNestingDepth > MAX_REPLICATION_NESTING_DEPTH)
       {
-      traceMsg(comp(), "for loop (%d), max nest depth thresholds exceeded\n", region->getNumber());
+      if (trace())
+         traceMsg(comp(), "for loop (%d), max nest depth thresholds exceeded\n", region->getNumber());
```

Comment:
I would argue that having this print even when not tracing the opt is very useful - it tells you if a loop has failed replication because of nesting depth. That can save reading a lot of detailed trace and is a generally useful diagnostic message.